src/main/java/io/jenkins/plugins/checks/github/GitSCMChecksContext.java

```diff
@@ -116,7 +131,7 @@ else if (repositoryUrl.matches("git@.+:.+\\/.+")) {
     @Override
     @CheckForNull
     protected String getCredentialsId() {
-        return getUserRemoteConfig().getCredentialsId();
+        return this.config.getCredentialsId();
```

Comment:
That is how I had configured it but that is what was also not working because the credentials were sticking. I can look into it in the future but for now I don't think it's worth debugging.